### DISCUSSION TITLE
Button Color

### DISCUSSION THREAD

**Discussion by SlavicBlin** [tupoy-ya/TupoyeMenu#64] [Q&A] [answered]
Any way to change the button color the same way as the GUI color?? i feel like that would be great for personalised Themes. Rn i do see that the buttons are different to command button colors so linking like that breaks it. any help would be great or if someone could figure it out and put it in their fork that would even better but i do understand that it takes time :)

**Comment by tupoy-ya** [comment, marked as answer]
There was recently a discussion about implementing a style editor in YimMenu matrix, however there is no timeline on this and it may never be implemented.

I can't really understand what exactly you are asking, but some buttons are colored with hard coded template colors which currently cannot be changed at runtime.

`src/gui/components/components.hpp`
```cpp
template<ImVec2 size = ImVec2(0, 0), ImVec4 color = ImVec4(0.24f, 0.23f, 0.29f, 1.00f)>
static bool button(const std::string_view text)
{
	bool status = false;
	ImGui::PushStyleColor(ImGuiCol_Button, color);
	status = ImGui::Button(text.data(), size);
	ImGui::PopStyleColor(1);
	return status;
}
```

Sorry if i didn't answer your question.

**Reply by SlavicBlin** [reply]
ahh i see yh i was just wondering if there was any way to change the hard coded template colors same way as the GUI color editor in settings but if there is no way to do it right now then thats fine :) Thank you for trying to understand what i was saying i just didnt really know how to word it but you got what i was trying to ask 👍
